Fix Expand/Collapse element disappears when dataTreeElementColumn is set and cell is editable

danielHin opened the pull request:
fixes the issue olifolkerd/tabulator#3173

When editing is stopped (canceled or successful) row.reinitialize was only called in case the index of the cell is 0. 
In case the dataTreeElementColumn is set to a specific field, the Expand/Collapse element wasn't re-added to the element after finishing editing. 

I've added a quick check if the field is set to be the dataTreeElementColumn when checkForRestyle is called. 

I'm not quite sure if I correctly accessed the options and the field property regarding your code style guide (if there is any function which should be used instead of directly accessing the property). 

Cheers
Daniel

olifolkerd commented:
Hey @mbellagamba

Im afraid we are not currently accepting PR's on the master branch.

The library has been completely rebuilt for the 5.0 release and most of the original files are no longer there.

If you are happy to resubmit this on the 5.0 branch i would be happy to consider it for merge.

Cheers

Oli :)

danielHin commented:
Hey @olifolkerd 

allright. Wasn't sure if there will be still bugfix releases for 4.

I will have a look into 5.0

- Daniel

danielHin commented:
@olifolkerd 
rebased and updated the PR.

I had to change two additional files to get tabulator running locally. You might already fixed that on other branches?

olifolkerd commented:
Hey @danielHin 

Thanks for the PR, this has already been fixed in a different way as part of the pre-release testing, but thanks for your input.

Version 5.0 will be released later today!

Cheers

Oli :)
